src/simulation.jl

```diff
@@ -84,6 +161,75 @@ end
 
 infer_state_names(d) = d[infer_states(d), :name]
 
+# This will be the function and matrix generation
+function compile_env(d::AbstractNamedDecapode, dec_matrices::Vector{Symbol})
```

Comment:
Git's diff algorithm thinks this whole function is new, but that's probably alright.